app/src/store/data.ts

```diff
@@ -302,8 +302,32 @@ export default function useDataStore() {
     const whitelistUrl = import.meta.env.VITE_GRANT_WHITELIST_URI;
     if (whitelistUrl) {
       const url = whitelistUrl + uniqueStr;
-      const json = await fetch(url).then((res) => res.json());
-      grants = grants.filter((grant) => json[DGRANTS_CHAIN_ID].includes(grant.id));
+
+      const filterWhiteListed = (whiteList: Record<number, [number]>) =>
```

Comment:
Two things here:
1. Nit: here we style variable as `whiteList` but above we use `whitelist`. I say we stick with `whitelist` everywhere for consistency
2. The type is defined as `Record<number, [number]>`, where `[number]` means a 1-element tuple where that element is a number. I think you want `number[]`, indicating an array of numbers